include/libtorrent/aux_/typed_span.hpp

```diff
@@ -49,6 +49,9 @@ namespace libtorrent { namespace aux {
 
 		// pull in constructors from base class
 		using base::base;
+		// implicit constructor needed for template based member functions
+		typed_span() = default;
+		typed_span(span<T> const& v) : base(v) {} // NOLINT
```

Comment:
I understand you, and this is an interesting edge case of the type system, because it goes to the root of how template member functions are implemented in template based types. I think I will try something else.